### ISSUE TITLE
auto_ossec on Windows

### ISSUE THREAD

**Issue by somepublicgithubact** [BinaryDefense/auto-ossec#9]
I cannot seem to get the Windows client to successfully register with the server.  The communication occurs, but I get the below error:

```
c:\Program Files (x86)\BinaryDefense\AutoOSSEC>auto_ossec.exe serverip *
[*] Connected to auto enrollment server at IP: serverip
[*] Pulled hostname and IP, encrypted data, and now sending to server.
[*] We received our new pairing key for OSSEC, closing server connection.
[*] Removing any old keys.
[*] Successfully imported the new pairing key.
[*] Stopping the OSSEC service, just in case its running.
[!] Unable to find the ossec.conf file in: /var/ossec/\ossec.conf
[!] Please install OSSEC first before running any of this.
```

Note: my OSSEC server is an AlienVault USM.  Not sure if the above error message is related to the client or the server, but on the server side I believe the ossec.conf is located in /var/ossec/etc.

Any suggestions would be really appreciated.  Auto-ossec seems like a wonderful solution to the pain of mass deployment.

**Comment by somepublicgithubact** [comment]
Seems that auto_ossec.exe is looking for files c:\var\ossec instead of C:\Program Files (x86)\ossec-agent

**Comment by somepublicgithubact** [comment]
For now, made simple short cut from c:\var\ossec > C:\Program Files (x86)\ossec-agent and seems to resolve the issue of where it is looking and writing files to.

On the server side, I am getting this output:
Client connected with  ('ipaddr', 56183)
Type str doesn't support the buffer API
Pairing complete. Terminating connection to client.

**Comment by somepublicgithubact** [comment]
Seems to happen during this: 

```
aes = DecodeAES(cipher, data)
->
DecodeAES = lambda c, e: c.decrypt(
                base64.b64decode(e)).rstrip(PADDING)
```

**Comment by somepublicgithubact** [comment]
Fixed with:
DecodeAES = lambda c, e: str(c.decrypt(
                base64.b64decode(e)), 'UTF-8').rstrip(PADDING)

New issue:
name 'file' is not defined

**Comment by somepublicgithubact** [comment]
Changed file("lock" to open("lock" for python3

New issue:
perhaps doesn't like the '*' for the IP of the client

```
command: /var/ossec/bin/manage_agents
args: ['/var/ossec/bin/manage_agents']
buffer (last 100 chars): b": *\r\n\r\n** Invalid IP '*'. Please enter a valid IP Address.\r\n\r\n   * The IP Address of the new agent: "
before (last 100 chars): b": *\r\n\r\n** Invalid IP '*'. Please enter a valid IP Address.\r\n\r\n   * The IP Address of the new agent: "
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 21525
child_fd: 8
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile("b'for the new agent'")
Pairing complete. Terminating connection to client.
[*] New OSSEC agent added - triggering restart of service to add..
```

**Comment by somepublicgithubact** [comment]
using 0.0.0.0/0 instead of *

**Comment by denied39** [comment]
@somepublicgithubact Did you have to install python3 for Alienvault? I just went back to an older auto_ossec because Alienvault doesn't have python3.

**Comment by somepublicgithubact** [comment]
I installed python3 on my Alienvault box to support auto_ossec.  I have finished with a number of conversions of 2to3 stuff on the server side, but now I am back to an issue client side.  

@denied39 - I might just do the same as you mentioned.

**Comment by somepublicgithubact** [comment]
Can you point me to a version in the history that you are successfully running?

**Comment by trustedsec** [comment]
What version of auto_ossec are you using? I can't reproduce this one..

C:\Users\Administrator\Desktop>auto_ossec.exe 192.168.170.130 *
[_] Connected to auto enrollment server at IP: 192.168.170.130
[_] Pulled hostname and IP, encrypted data, and now sending to server.
[_] We received our new pairing key for OSSEC, closing server connection.
[_] Removing any old keys.
[_] Successfully imported the new pairing key.
[_] Stopping the OSSEC service, just in case its running.
[_] Overwriting the ossec.conf to incorporate server host IP address.
[_] Finished. Started the OSSEC service. Auto Enrollment for OSSEC is now finished.

As far as python3 - there's a couple of issues here, I've changed file to open which is easy - AES is a lot different as far as formatting. I started digging into python3 support it will just take some time for full conversion. 2to3 doesn't work - it's more complex then that.

**Comment by trustedsec** [comment]
Figured out your time delay issue - hostname is not stripping the padding:

hostname{{{{{{{{

Working on fixing it now

**Comment by trustedsec** [comment]
More than that - looks like new OSSEC server doesn't allow wildcards now

**Comment by trustedsec** [comment]
Remoted doesn't allow it either:

2016/07/22 01:49:12 ossec-remoted(1237): ERROR: Invalid ip address: '*'.

**Comment by somepublicgithubact** [comment]
You can use 0.0.0.0/0
That is, if I send in a \* I changed the server side to change that to 0.0.0.0/0
If the client doesn't send in the \* then it should take the IP of the client like you had it still.

**Comment by somepublicgithubact** [comment]
I did a bunch of conversions for strings and bytes...I can post my files tomorrow if you like.  Let me clean them up.  I just got it working now on my environment.  Running python3 on both client and server.

**Comment by trustedsec** [comment]
Super nice. Thanks! I just pushed the latest version without the 0.0.0.0 stuff but got python2/3 working.

**Comment by trustedsec** [comment]
I'll get 0.0.0.0 working

**Comment by trustedsec** [comment]
okay fixed on all fronts, git pull the latest and let me know if you run into any issues on auto_server.py working on python3 - also star certificate working now too.. I don't see the auto_ossec.exe issue - any ideas on how to reproduce that one?

**Comment by somepublicgithubact** [comment]
I still get a lot of str vs. byte issues...here is the first one after adding a traceback on auto_ossec.py

```
>python auto_ossec.py localip *
[*] Connected to auto enrollment server at IP: localip
[*] Pulled hostname and IP, encrypted data, and now sending to server.
[*] Something did not complete. Does this system have Internet access?
Traceback (most recent call last):
  File "auto_ossec.py", line 335, in <module>
    s.send(data)
TypeError: a bytes-like object is required, not 'str'
```

related to:
data being of type str and the socket wanting bytes.
`s.send(data)`

I converted a bunch of spots like this to resolve the issue.

**Comment by somepublicgithubact** [comment]
Posted my changes to auto_ossec.py and auto_server.py (https://github.com/somepublicgithubact/auto-ossec) for your consideration.  

Note: some of it is not written very pythonic since I was decomposing some of it to better understand how it works.  
Note2: tested client side on Windows and Server side on Alienvault distro of OSSEC....both with python3

**Comment by trustedsec** [comment]
Can't thank you enough for working on this. Did help out significantly. It didn't get all the way through if you use python2 - was able to rewrite portions and add python2 support so this should work on python2 and python3, I'll leave this open for now - let me know if you have any issues - if you don't I'll close this in a couple days. Really appreciate it!

**Comment by trustedsec** [comment]
I'll be committing here in a few minutes just packaging the binaries now.

**Comment by trustedsec** [comment]
Alright, just pushed it, thanks again!

**Comment by trustedsec** [comment]
Just pushed version 1.6 - spent a lot of time adding queues as well

**Comment by somepublicgithubact** [comment]
Works great!  Thank you much.

**Comment by miraclesoul** [comment]
Hello guys,

i am having all the above same issues. it worked well on windows 7 agent machine but now running the same on Windows 8.1 and getting the following errors:
------------------------------------------------------------------
>python auto_ossec.py localip *
[*] Connected to auto enrollment server at IP: localip
[*] Pulled hostname and IP, encrypted data, and now sending to server.
[*] Something did not complete. Does this system have Internet access?
------------------------------------------------------------------
C:\Users\fahmad\Downloads\auto-ossec-master\auto-ossec-master>auto_ossec.exe 10.
10.100.12
[*] Connected to auto enrollment server at IP: 10.10.100.12
[*] Pulled hostname and IP, encrypted data, and now sending to server.
[*] We received our new pairing key for OSSEC, closing server connection.
[*] Removing any old keys.
[*] Something did not complete. Does this system have Internet access?
Traceback (most recent call last):
  File "auto_ossec.py", line 356, in <module>
  File "base64.py", line 78, in b64decode
TypeError: Incorrect padding
None
------------------------------------------------------------------

Can I have an updated final version of the file for windows. I see that there have been updates but i need the final thing to run and deploy on over 100 servers.

Thanks

**Comment by gwscobie** [comment]
Windows server 2012 r2, i randomly receive the following error below
Traceback (most recent call last):
  File "auto_ossec.py", line 356, in <module>
  File "base64.py", line 78, in b64decode
TypeError: Incorrect padding
None

Any ideas
